components/List/src/MDCSelfSizingStereoCell.m

```diff
@@ -207,24 +210,45 @@ - (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
                                                   object:nil];
   }
 
-  [self adjustFontsForContentSizeCategory];
+  [self adjustFontsForDynamicType];
+}
+
+- (void)mdc_setLegacyFontScaling:(BOOL)mdc_legacyFontScaling {
+  if (mdc_legacyFontScaling == _mdc_legacyFontScaling) {
+    return;
+  }
+
+  _mdc_legacyFontScaling = mdc_legacyFontScaling;
+
+  [self adjustFontsForDynamicType];
 }
 
 // Handles UIContentSizeCategoryDidChangeNotifications
 - (void)contentSizeCategoryDidChange:(__unused NSNotification *)notification {
-  [self adjustFontsForContentSizeCategory];
+  [self adjustFontsForDynamicType];
 }
 
-- (void)adjustFontsForContentSizeCategory {
+- (void)adjustFontsForDynamicType {
   UIFont *titleFont = self.titleLabel.font ?: self.defaultTitleLabelFont;
   UIFont *detailFont = self.detailLabel.font ?: self.defaultDetailLabelFont;
-  if (_mdc_adjustsFontForContentSizeCategory) {
-    titleFont =
-        [titleFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleTitle
-                                scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
-    detailFont =
-        [detailFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleCaption
-                                 scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    if (self.mdc_legacyFontScaling) {
+      titleFont =
+          [titleFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleTitle
+                                  scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+      detailFont =
+          [detailFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleCaption
+                                   scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+    } else if (titleFont.mdc_scalingCurve && detailFont.mdc_scalingCurve) {
```

Comment:
I would expect these fonts to be scaled independently - why are we checking for the scaling curves?

Also, the Dynamic Type methods are "safe" - if there are no curves attached, they return `self` so there would be no need to include these checks at all.

Reply:
I think it was originated from the pattern in MDCButton. I assume it serves as an early guard. I don't feel strongly on it (https://github.com/material-components/material-components-ios/pull/6947#discussion_r269629838) but we need to match the pattern across the components once a decision is made.

Reply:
I don't see MDCButton checking 2 different fonts to ensure they're both scalable - this seems to be new to Lists. Do we have a specific reason for making this change?

Reply:
I was more thinking of 

> why are we checking for the scaling curves?

MDCButton checks one font `font.mdc_scalingCurve` to ensure it is scalable. For 2 fonts, it could be the scope of another discussion.

Reply:
Let's keep it in this thread. We don't need to check these implementation details (the code already does this), and we shouldn't be checking for both before using either.

Reply:
Followed up offline. I agree it is a common use case that only title is set up and needs to be scalable. In this case, we should add that code path.